src/Entity/EntityIdValue.php

```diff
@@ -31,8 +31,8 @@ public function __construct( EntityId $entityId ) {
 	 */
 	public function serialize() {
 		return json_encode( [
-			$this->entityId->getEntityType(),
-			$this->getNumericId()
+			get_class( $this->entityId ),
```

Comment:
Oh, I definitely do not want this in the database, or in our external representation! As far as I am aware, this is not used in any way with our JSON encoding. We only use the php-serialization temporarily, to do deep clones. PHP-Serialization is extremely brittle (sensitive against changes to private members, etc), so it should never be used for persistence.

The alternative would be to change the way we do cloning, and avoid copying immutable objects when we clone. That would be nice, but I can't think of a reliable and easy way of doing this. serialize/unserialize is a very convenient method to do deep cloning safely. But it's somewhat wasteful, since it also copies immutable objects.